### PR TITLE
feat(eip/associate): support to associate an EIP with network_id and fixed_ip

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

* support to associate an EIP with network_id and fixed_ip
* add mac_address and status attributes
*  fix an error when importing an unbound EIP

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST='./huaweicloud/services/acceptance/eip' TESTARGS='-run TestAccEIPAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eip -v -run TestAccEIPAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccEIPAssociate_basic
=== PAUSE TestAccEIPAssociate_basic
=== CONT  TestAccEIPAssociate_basic
--- PASS: TestAccEIPAssociate_basic (99.23s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eip       99.305s

$ make testacc TEST='./huaweicloud/services/acceptance/eip' TESTARGS='-run TestAccEIPAssociate_port'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eip -v -run TestAccEIPAssociate_port -timeout 360m -parallel 4
=== RUN   TestAccEIPAssociate_port
=== PAUSE TestAccEIPAssociate_port
=== CONT  TestAccEIPAssociate_port
--- PASS: TestAccEIPAssociate_port (97.10s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eip       97.166s
```

### CODE REVIEW

File: huaweicloud/services/eip/resource_huaweicloud_eip_associate.go [hunk 1]
```diff
@@ -37,18 +42,41 @@ func ResourceEIPAssociate() *schema.Resource {
...
-
-			"port_id": {
+			"public_ip": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
-
-			"public_ip": {
+			"port_id": {
```
Comment (niuzhenguo) [inline huaweicloud/services/eip/resource_huaweicloud_eip_associate.go:50]: Suggest to add Conflicts between `port_id` and `fixed_ip`.

Comment (Jason-Zhang9309) [reply huaweicloud/services/eip/resource_huaweicloud_eip_associate.go:50]: ExactlyOneOf could be better

File: huaweicloud/services/eip/resource_huaweicloud_eip_associate.go [hunk 2]
```diff
@@ -37,18 +42,41 @@ func ResourceEIPAssociate() *schema.Resource {
...
+				ForceNew:     true,
+				ValidateFunc: validation.IsIPAddress,
+				AtLeastOneOf: []string{"port_id"},
+				RequiredWith: []string{"network_id"},
+			},
+			"network_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/eip/resource_huaweicloud_eip_associate.go]: suggest to add RequiredWith: []string{"fixed_ip"} beacuse network_id is useless when fixed_ip is not specified

File: huaweicloud/services/eip/resource_huaweicloud_eip_associate.go [hunk 3]
```diff
@@ -86,12 +125,39 @@ func resourceEIPAssociateRead(_ context.Context, d *schema.ResourceData, meta in
...
+		d.Set("port_id", eIP.PortID),
+		d.Set("public_ip", eIP.PublicAddress),
+		d.Set("fixed_ip", eIP.PrivateAddress),
+		d.Set("network_id", associatedPort.NetworkId),
+		d.Set("mac_address", associatedPort.MacAddress),
+		d.Set("status", NormalizeEIPStatus(eIP.Status)),
+	)
+
+	if err = mErr.ErrorOrNil(); err != nil {
+		return fmtp.DiagErrorf("Error setting eip fields: %s", err)
```
Comment (hmbig2) [inline huaweicloud/services/eip/resource_huaweicloud_eip_associate.go]: s/eip/eip associate

File: huaweicloud/services/acceptance/eip/resource_huaweicloud_eip_associate_test.go [hunk 4]
```diff
@@ -16,6 +16,7 @@ func TestAccEIPAssociate_basic(t *testing.T) {
 	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/eip/resource_huaweicloud_eip_associate_test.go]: s/fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))/acceptance.RandomAccResourceNameWithDash()

File: huaweicloud/services/acceptance/eip/resource_huaweicloud_eip_associate_test.go [hunk 5]
```diff
@@ -34,6 +35,48 @@ func TestAccEIPAssociate_basic(t *testing.T) {
...
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccEIPAssociate_port(t *testing.T) {
+	var eip eips.PublicIp
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	associateName := "huaweicloud_vpc_eip_associate.test"
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/eip/resource_huaweicloud_eip_associate_test.go:58]: s/fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))/acceptance.RandomAccResourceNameWithDash()

Comment (ShiChangkuo) [reply huaweicloud/services/acceptance/eip/resource_huaweicloud_eip_associate_test.go:58]: fixed

File: huaweicloud/services/acceptance/eip/resource_huaweicloud_eip_associate_test.go (hunk 4 above)
Comment (ShiChangkuo) [reply huaweicloud/services/acceptance/eip/resource_huaweicloud_eip_associate_test.go]: fixed

File: huaweicloud/services/eip/resource_huaweicloud_eip_associate.go (hunk 3 above)
Comment (ShiChangkuo) [reply huaweicloud/services/eip/resource_huaweicloud_eip_associate.go]: fixed

File: huaweicloud/services/eip/resource_huaweicloud_eip_associate.go (hunk 1 above)
Comment (ShiChangkuo) [reply huaweicloud/services/eip/resource_huaweicloud_eip_associate.go:50]: **ExactlyOneOf**  is updated

File: huaweicloud/services/eip/resource_huaweicloud_eip_associate.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/services/eip/resource_huaweicloud_eip_associate.go]: **ConflictsWith** has updated

File: huaweicloud/services/eip/resource_huaweicloud_eip_associate.go [hunk 6]
```diff
@@ -155,3 +222,39 @@ func actionOnPort(client *golangsdk.ServiceClient, eipID, portID string, timeout
...
+					portID = item.ID
+					return false, nil
+				}
+			}
+		}
+		return true, nil
+	})
+
+	if err != nil {
+		return "", fmtp.Errorf("Unable to list ports: %s", err)
```
Comment (Lance52259) [inline huaweicloud/services/eip/resource_huaweicloud_eip_associate.go:252]: first letter lowercase.

**Pull request merged**